src/Aspire.Hosting.Azure/AzureCosmosDBResource.cs

```diff
@@ -22,5 +24,43 @@ public class AzureCosmosDBResource(string name, string? connectionString)
     /// Gets the connection string to use for this database.
     /// </summary>
     /// <returns>The connection string to use for this database.</returns>
-    public string? GetConnectionString() => ConnectionString;
+    public string? GetConnectionString() => IsEmulator
+        ? AzureCosmosDBEmulatorConnectionString.Create(GetEmulatorPort("emulator"))
+        : ConnectionString;
+
+    /// <summary>
+    /// Gets a value indicating whether the Azure Cosmos DB resource is running in the local emulator.
+    /// </summary>
+    public bool IsEmulator => this.IsContainer();
+
+    private int GetEmulatorPort(string endpointName) =>
+        Annotations
+            .OfType<AllocatedEndpointAnnotation>()
```

Comment:
Is this still AllocatedEndpointAnnotation or ServiceBindingAnnotation?

Reply:
Good catch, I was borrowing the pattern from here `AzureStorageResource` bits: 

https://github.com/dotnet/aspire/blob/fc52d4a8c7891c5a3a3d3bfd663940b9f99f0600/src/Aspire.Hosting.Azure/AzureStorageResource.cs#L47-L52

And I see that it adds `ServiceBindingAnnotation`, not `AllocatedEndpointAnnotation`:

https://github.com/dotnet/aspire/blob/fc52d4a8c7891c5a3a3d3bfd663940b9f99f0600/src/Aspire.Hosting.Azure/AzureResourceExtensions.cs#L161-L167

I can change this, but I'd like to understand how the other one works. Is there something that updates the annotations collection afterwards?